modules/apps/collaboration/message-boards/message-boards-service/src/main/java/com/liferay/message/boards/internal/exportimport/data/handler/MBDiscussionStagedModelDataHandler.java

```diff
@@ -212,7 +214,20 @@ private StagedModel _getReferencedStagedModel(MBDiscussion discussion) {
 				return null;
 			}
 
-			return (StagedModel)persistedModel;
+			StagedModel stagedModel = (StagedModel)persistedModel;
+
+			StagedModelDataHandler<?> stagedModelDataHandler =
+				StagedModelDataHandlerRegistryUtil.getStagedModelDataHandler(
+					stagedModel.getModelClassName());
+
+			if (persistedModel instanceof GroupedModel) {
+				GroupedModel groupedModel = (GroupedModel)persistedModel;
+
+				return stagedModelDataHandler.fetchStagedModelByUuidAndGroupId(
+					stagedModel.getUuid(), groupedModel.getGroupId());
+			}
+
+			return null;
```

Comment:
Shouldn't this be `return persistedModel;` to be consistent with the previous implementation?